safe_transaction_service/contracts/management/commands/update_safe_contracts_logo.py

```diff
@@ -0,0 +1,71 @@
+from django.core.files import File
+from django.core.management import BaseCommand, CommandError
+
+from gnosis.eth import EthereumClientProvider
+from gnosis.safe.safe_deployments import safe_deployments
+
+from safe_transaction_service.contracts.models import Contract
+
+
+def generate_safe_contract_display_name(contract_name: str, version: str) -> str:
+    """
+    Generates the display name for Safe contract.
+    Append Safe at the beginning if the contract name doesn't contain Safe word and append the contract version at the end.
+
+    :param contract_name:
+    :param version:
+    :return: display_name
+    """
+    if "safe" not in contract_name.lower():
+        return f"Safe: {contract_name} {version}"
+    else:
+        return f"{contract_name} {version}"
+
+
+class Command(BaseCommand):
+    help = "Update or create Safe contracts with provided logo"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--safe-version", type=str, help="Contract version", required=False
+        )
+        parser.add_argument(
+            "--logo-path", type=str, help="Path of new logo", required=True
+        )
+
+    def handle(self, *args, **options):
+        """
+        Command to create or update Safe contracts with provided logo.
+
+        :param args:
+        :param options: Safe version and logo path
+        :return:
+        """
+        safe_version = options["safe_version"]
+        logo_path = options["logo_path"]
+        ethereum_client = EthereumClientProvider()
+        chain_id = ethereum_client.get_chain_id()
+        logo_file = File(open(logo_path, "rb"))
+        if not safe_version:
+            versions = list(safe_deployments.keys())
+        elif safe_version in safe_deployments:
+            versions = [safe_version]
+        else:
+            raise CommandError(
+                f"Wrong Safe version {safe_version}, supported versions {safe_deployments.keys()}"
+            )
+
+        for version in versions:
+            for contract_name, addresses in safe_deployments[version].items():
+                contract, created = Contract.objects.get_or_create(
+                    address=addresses[str(chain_id)],
+                    defaults={
```

Comment:
A lot of contracts are already created, so their name will remain empty, so maybe we should use `update_or_create`?

Reply:
I was using get_or_create because I was trying to don't update the name or display_name of contracts are already created. 
I agree with update every contract and contract name as follows: 
- contract name = contract name on safe deployments
- display name = ('Safe: '+contractNameWithoutSafeWord+contractVersion') or (contractName+contractVersion) 
For example for display name of MultiSend and SafeL2 for 1.4.1 contracts, would be:
- Safe: MultiSend 1.4.1 
- SafeL2 1.4.1

Reply:
I just added a flag to decide is all Safe contract names are updated or only by default the contract names that are empty.
I can`t find a way to do this only with `update_or_create` adding conditional Case update, because broke when check if name is empty and the row doesn't exist.